grpc/optimistic/server.go

```diff
@@ -0,0 +1,308 @@
+		case err := <-mempoolClearingEvent.Err():
+			if err != nil {
+				log.Error("error waiting for mempool clearing event", "err", err)
+				return status.Errorf(codes.Internal, shared.WrapError(err, "error waiting for mempool clearing event").Error())
+			} else {
+				// TODO - what is the right error code here?
+				return status.Error(codes.Internal, "mempool clearance subscription closed")
+			}
+		case <-stream.Context().Done():
+			return stream.Context().Err()
+		}
+	}
+}
+
+func (o *AuctionServiceV1Alpha1) ExecuteOptimisticBlock(ctx context.Context, req *auctionPb.BaseBlock) (*astriaPb.Block, error) {
+	// we need to execute the optimistic block
+	log.Debug("ExecuteOptimisticBlock called", "timestamp", req.Timestamp, "sequencer_block_hash", common.BytesToHash(req.SequencerBlockHash).String())
+
+	// Deliberately called after lock, to more directly measure the time spent executing
+	executionStart := time.Now()
+	defer executionOptimisticBlockTimer.UpdateSince(executionStart)
+
+	if err := validateStaticExecuteOptimisticBlockRequest(req); err != nil {
+		log.Error("ExecuteOptimisticBlock called with invalid BaseBlock", "err", err)
+		return nil, status.Error(codes.InvalidArgument, shared.WrapError(err, "invalid BaseBlock").Error())
+	}
+
+	if !o.SyncMethodsCalled() {
+		return nil, status.Error(codes.PermissionDenied, "Cannot execute block until GetGenesisInfo && GetCommitmentState methods are called")
+	}
+
+	softBlock := o.Bc().CurrentSafeBlock()
+
+	nextFeeRecipient := o.NextFeeRecipient()
+
+	// the height that this block will be at
+	height := o.Bc().CurrentBlock().Number.Uint64() + 1
+
+	addressPrefix := o.Bc().Config().AstriaSequencerAddressPrefix
+
+	txsToProcess := shared.UnbundleRollupDataTransactions(req.Transactions, height, o.BridgeAddresses(), o.BridgeAllowedAssets(), softBlock.Hash().Bytes(), o.AuctioneerAddress(), o.AuctioneerStartHeight(), addressPrefix)
+
+	// Build a payload to add to the chain
+	payloadAttributes := &miner.BuildPayloadArgs{
+		Parent:                softBlock.Hash(),
+		Timestamp:             uint64(req.GetTimestamp().GetSeconds()),
+		Random:                common.Hash{},
+		FeeRecipient:          nextFeeRecipient,
+		OverrideTransactions:  txsToProcess,
+		IsOptimisticExecution: true,
+	}
+	payload, err := o.Eth().Miner().BuildPayload(payloadAttributes)
+	if err != nil {
+		log.Error("failed to build payload", "err", err)
+		return nil, status.Errorf(codes.InvalidArgument, shared.WrapError(err, "failed to build payload").Error())
+	}
+
+	block, err := engine.ExecutableDataToBlock(*payload.Resolve().ExecutionPayload, nil, nil)
+	if err != nil {
+		log.Error("failed to convert executable data to block", err)
+		return nil, status.Error(codes.Internal, shared.WrapError(err, "failed to convert executable data to block").Error())
+	}
+
+	// this will insert the optimistic block into the chain and persist it's state without
```

Comment:
```suggestion
	// this will insert the optimistic block into the chain and persist its state without
```

Reply:
addressed in https://github.com/astriaorg/flame/pull/30/commits/defc35bb0c32331a823a635dccdb28c7e62d519b